### PR TITLE
Business UI

### PR DESCRIPTION
**Pull request by p-janik**
https://linear.app/ageras-group/issue/KRP-1210/%5Bms%5D-create-business-account-on-mock-solaris

Screenshot 2024-11-04 at 15 12 10

### CODE REVIEW

**Review by lera — DISMISSED** [review]

File: src/templates/business.html [hunk 1]
```diff
@@ -13,28 +14,163 @@ <h2>{{ business.name }}</h2>
...
+            <input type="text" class="form-control" name="createdAt" id="createdAt" placeholder="Created At"
+              value="{{ business.createdAt }}" disabled>
           </div>
           <button type="submit" class="btn btn-primary btn-block">Submit</button>
         </form>
       </div>
     </div>
+
+    <!-- Address -->
+    <div class="panel panel-default">
```
Comment (p-janik) [inline src/templates/business.html:39]: ops, should i add update logic? 😅

File: src/routes/accountOpeningRequest.ts [hunk 2]
```diff
@@ -158,21 +146,23 @@ export const retrieveAccountOpeningRequest = async (
     let customerType = CustomerType.PERSON;
     let entityId: string;
     let entity: MockPerson | MockBusiness;
+    let handlers = getHandlers(customerType);
```
Comment (MikolajKaminski) [inline src/routes/accountOpeningRequest.ts]: What if we were to move the getHandler into the if/else statement so it's only called once 😄 

```suggestion
    let handlers;
```

File: src/routes/accountOpeningRequest.ts [hunk 3]
```diff
@@ -158,21 +146,23 @@ export const retrieveAccountOpeningRequest = async (
...
 
     entityId = await getCustomerIdByAccountOpeningRequest(
       accountOpeningRequestId,
       customerType
     );
 
     if (entityId) {
-      entity = await getPerson(entityId);
+      entity = await handlers.getEntity(entityId);
```
Comment (MikolajKaminski) [inline src/routes/accountOpeningRequest.ts]: Followup to previous suggestion

```suggestion
      handlers = getHandlers(customerType);
      entity = await handlers.getEntity(entityId);
```

File: src/templates/business.html [hunk 4]
```diff
@@ -13,28 +14,159 @@ <h2>{{ business.name }}</h2>
...
+        {% for representative in business.legalRepresentatives %}
+        <div class="form-group">
+          <p><strong>Representative ID:</strong> {{ representative.legal_representative_id }}</p>
+          <p><strong>Type of Representation:</strong> {{ representative.type_of_representation }}</p>
+          <p><strong>Power of Attorney Confirmed At:</strong> {{ representative.power_of_attorney_confirmed_at }}</p>
+        </div>
+        <hr>
+        {% endfor %}
+      </div>
+    </div>
```
Comment (MikolajKaminski) [inline src/templates/business.html]: That's a little nitpicky, but wouldn't it make sense for the `LR's` to be above the `BO's`?

**Review by MikolajKaminski — APPROVED** [review]
Looks good to me 👍 
Small suggestions only

**Comment by lera** [comment]
would be cool to have a link to the person resource of LR, to succeed identification. But it also can be different ticket 😉

**Pull request merged**